cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml

```diff
@@ -667,6 +677,211 @@ Resources:
       MinSize: !Ref AsgWarmPoolMinSize
       PoolState: !Ref AsgWarmPoolState
 
+  # ASG Lambda Related resources
+
+  CcCloudwatchLogGroupForLambda:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub '${AWS::StackName}-CcCloudwatchLogGroupForLambda'
+
+  CcGetSecretsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Get Secrets Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'secretsmanager:GetSecretValue'
+
+  CcCloudwatchMetricsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Cloudwatch Metrics Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'cloudwatch:GetMetricStatistics'
+              - 'cloudwatch:ListMetrics'
+              - 'cloudwatch:PutMetricData'
+              - 'ec2:DescribeTags'
+            Resource:
+              - '*'
+
+  CcCloudwatchLogsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Cloudwatch Logs Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'logs:CreateLogStream'
+              - 'logs:PutLogEvents'
+            Resource:
+              - !GetAtt 
+                - CcCloudwatchLogGroupForLambda
+                - Arn
+          - Effect: Allow
+            Action:
+              - 'logs:CreateLogGroup'
+            Resource:
+              - !Join
+                - ':'
+                - - arn
+                  - aws
+                  - 'logs:'
+                  - !Sub '${AWS::Region}'
+                  - !Sub '${AWS::AccountId}'
+                  - '*'
+
+  CcAutoScalingLifecycleActionsPolicyForLambda:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Zscaler Cloud Connector Cloudwatch Metrics Policy For Lambda
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'autoscaling:DescribeLifecycleHookTypes'
+              - 'autoscaling:DescribeLifecycleHooks'
+              - 'autoscaling:DescribeAutoScalingInstances'
+              - 'autoscaling:CompleteLifecycleAction'
+              - 'autoscaling:RecordLifecycleActionHeartbeat'
+              - 'autoscaling:SetInstanceHealth'
+              - 'ec2:DescribeInstanceStatus'
```

Comment:
Please add
                "ec2:DescribeInstances",
                "ec2:DescribeTags",

Reply:
are you asking for both asg policy and asg policy for lambda to add these two actions ?

Reply:
just asg policy for lambda only, please ignore the other